src/kixi/nhs/data/transform.clj

```diff
@@ -107,6 +107,24 @@
   [data recipe]
   (update-in data [:value] process-percentages))
 
+;; Calculate confidence intervals:
+;; when several time periods in
+;; the same data resource in ckan
+(defn calculate-lci-uci
+  [condition data]
+  (if condition
+    (let [sorted-data (reverse (sort-by :end_date data))
+          current (if (not (nil? (:value (first sorted-data))))
+                    (Float/parseFloat (:value (first sorted-data))))
+          previous (if (not (nil? (:value (second sorted-data))))
+                     (Float/parseFloat (:value (second sorted-data))))
+          difference (- current previous)
+          ci {:lci (if (not (nil? difference)) (str (- current difference)))
+              :uci (if (not (nil? difference)) (str (+ current difference)))}]
+      (into [] (concat [(merge (first sorted-data) ci)]
+                       (into [] (next sorted-data)))))
+    data))
+
```

Comment:
```clojure
(defn calculate-lci-uci
        [condition data]
        (if condition
          (let [sorted-data (reverse (sort-by :end_date data))
                current     (transform/parse-number (:value (first sorted-data)))
                previous    (transform/parse-number (:value (second sorted-data)))]
            (if (and current previous)
              (let [difference (- current previous)
                    ci         {:lci (if (not (nil? difference)) (str (- current difference)))
                                :uci (if (not (nil? difference)) (str (+ current difference)))}]
                (into [] (concat [(merge (first sorted-data) ci)]
                                 (into [] (next sorted-data)))))
              data))))
```
Something like this? Prevents NPE for subtraction as well, and reuses `parse-number` that we have in `transform.clj`. Not tested though. Just an example.